main/src/com/google/refine/operations/row/RowKeepMatchedOperation.java

```diff
@@ -67,7 +71,7 @@ public boolean visit(Project project, int rowIndex, Row row) {
         return new HistoryEntry(
                 historyEntryID,
                 project,
-                row_keep_matching_brief() + " - " + rowsToRemove.size() + " rows removed",
+                row_keep_matching_brief() + " - "+createDescription(project, rowsToRemove.size()),
```

Comment:
That is sadly still not enough. The " - " you have inserted between the two messages might not make sense in all languages, so it is better to have a single message which spans over the whole description.

The message could be:
`Keep only matching rows ({0} rows removed)`

This makes it possible to give full control to the translators over how to translate the entire string in their language.